combine/combine.go

```diff
@@ -0,0 +1,514 @@
+		return nil, nil, err
+	}
+
+	if err = cmd.Start(); err != nil {
+		return nil, nil, err
+	}
+
+	if err = sendFilePathsToSort(sortStdin, inputs); err != nil {
+		return nil, nil, err
+	}
+
+	return sortMergeOutput, cmd.Wait, nil
+}
+
+// sendFilePathsToSort will pipe the given paths null terminated to the pipe.
+// For use with the StdinPipe of an exec.Command for `sort -m --files0-from -`.
+// The in is closed afterwards.
+func sendFilePathsToSort(in io.WriteCloser, paths []string) error {
+	for _, path := range paths {
+		if _, err := in.Write([]byte(path + string(rune(0)))); err != nil {
+			return err
+		}
+	}
+
+	return in.Close()
+}
+
+// openFiles opens the given files for reading.
+func OpenFiles(paths []string) ([]*os.File, error) {
+	files := make([]*os.File, len(paths))
+
+	for i, path := range paths {
+		file, err := os.Open(path)
+		if err != nil {
+			return files, err
+		}
+
+		files[i] = file
+	}
+
+	return files, nil
+}
+
+// THIS FUNCTION NEEDS TO BE GENERALISED TO ALL FILES, NOT EXCLUSIVELY PGZIP.
+func Concatenate(inputs []*os.File, output *pgzip.Writer) error {
+	buf := make([]byte, bytesInMB)
+
+	for _, input := range inputs {
+		if _, err := io.CopyBuffer(output, input, buf); err != nil {
+			return err
+		}
+
+		if err := input.Close(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// CompressAndConcatenate compresses and concatenates the inputs and stores in
+// the output.
+func CompressAndConcatenate(sourceDir string, outputFile string) error {
+	paths, err := findFilePathsInDir(sourceDir, statOutputFileSuffix)
+	if err != nil {
+		return err
+	}
+
+	inputs, err := OpenFiles(paths)
+	if err != nil {
+		return err
+	}
+
+	output, err := createOutputFileInDir(sourceDir, filepath.Base(outputFile))
+	if err != nil {
+		return err
+	}
+
+	compressedOutput, closeOutput, err := Compress(output)
+	if err != nil {
+		return err
+	}
+
+	if err = Concatenate(inputs, compressedOutput); err != nil {
+		return err
+	}
+
+	closeOutput()
+
+	return nil
+}
+
+// Compress wraps the given output to compress data copied to it, and
+// returns the writer. Also returns a function that you should call to close
+// the writer and output when you're done.
+func Compress(output *os.File) (*pgzip.Writer, func(), error) {
+	compressedOutput := pgzip.NewWriter(output)
+
+	err := compressedOutput.SetConcurrency(bytesInMB, runtime.GOMAXPROCS(0)*pgzipWriterBlocksMultiplier)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return compressedOutput, func() {
+		err = compressedOutput.Close()
+		if err != nil {
+			return
+		}
+
+		err = output.Close()
+		if err != nil {
+			return
+		}
+	}, err
+}
+
+// mergeUserGroupStreamToCompressedFile merges pre-sorted (pre-merged) usergroup
+// data (eg. from a `sort -m` of .byusergroup files), summing consecutive lines
+// with the first 3 columns, and outputting the results to a file, compressed.
+func mergeUserGroupStreamToCompressedFile(data io.ReadCloser, output *os.File) error {
```

Comment:
This should be left in cmd/combine for now, until we decide what to do about filetype-specific merges.

Reply:
Remove it, along with the associated constants.